resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -383,13 +438,28 @@ protected void initializeRegistriesAndFilters()
       if (stringConverters == null && parent != null) return parent.getStringConverters();
       return stringConverters;
    }
-   protected List<ParamConverterProvider> getParamConverterProviders()
+   
+   public List<ParamConverterProvider> getParamConverterProviders()

```

Comment:
This method and state it maintains (paramConverterProviders) look obsolete.

Reply:
@sermojohn, that's one of the methods I included to maintain API compatibility.  I changed your method
```
protected Set<SortedKey<ParamConverterProvider>> getParamConverterProviders()
```
to
```
protected Set<SortedKey<ParamConverterProvider>> getSortedParamConverterProviders()
```
and restored the original
```
   public List<ParamConverterProvider> getParamConverterProviders()
```

Reply:
@ronsigal I do not get the meaning of being backwards compatible on public methods that are not part of the interfaces that the class implements. I would assume that these methods (like getExceptionMappers() and getParamConverterProviders()) should never be called from another class, if the "programming to an interface" principle. What do you think?

Reply:
@sermojohn, well, maybe "API compatibility" isn't quite the right term. But, technically, public and even protected methods are accessible to Resteasy users. However inadvisable, someone COULD be calling getParamConverterProviders(). All things being equal, I would like to prevent breaking someone's code unless there's a good reason.

Reply:
@sermojohn while I agree in principle with you, what @ronsigal is doing is unfortunately needed at least till we go and rework the way RESTEasy is split into components and integrated in WildFly. That's https://issues.jboss.org/browse/RESTEASY-1533 btw, but I'm not very confident we'll be able to work on that in RESTEasy 4.